witness/src/Dscp/CommonCLI.hs

```diff
@@ -94,13 +92,8 @@ appDirParamParser = AppDirectorySpecific <$>
                     pure AppDirectoryOS
 
 -- | Parses time with specified unit of measurement, e.g. @10s@.
-timeReadM :: KnownRat unit => ReadM (Time unit)
-timeReadM = asum
-    [ toUnit @_ @Second <$> auto
-    , toUnit @_ @Millisecond <$> auto
-    , toUnit @_ @Microsecond <$> auto
-    , toUnit @_ @Minute <$> auto
-    ]
+timeReadM :: KnownRatName unit => ReadM (Time unit)
+timeReadM = nothingToFail "Cannot parse time unit" . unitsP =<< str
```

Comment:
http://hackage.haskell.org/package/optparse-applicative-0.14.2.0/docs/Options-Applicative.html#v:maybeReader
But I already used it in #284 :stuck_out_tongue:

Reply:
Yeah now I see it, your version is better ))